server/src/main/java/org/opensearch/search/profile/query/QueryProfiler.java

```diff
@@ -104,4 +104,11 @@ public CollectorResult getCollector() {
         return collector.getCollectorTree();
     }
 
+    /**
+     * Return the top level breakdown for plugins to use
+     */
+    public ContextualProfileBreakdown getTopBreakdown() {
```

Comment:
This is used in tests only, right?

```suggestion
    ContextualProfileBreakdown getTopBreakdown() {
```

Reply:
No, this is used by plugins to get the current top query profile breakdown so that it can get the added `customProfileMetrics` and use the profile metrics.

Reply:
How? Could you please give me an example?